rich/pretty.py

```diff
@@ -83,7 +83,16 @@ def display_hook(value: Any) -> None:
             )
             builtins._ = value  # type: ignore
 
-    sys.displayhook = display_hook
+    try:  # pragma: no cover
+        ip = get_ipython()  # type: ignore
+        from IPython.core.formatters import BaseFormatter
+
+        # replace plain text formatter with rich formatter
+        formatter = BaseFormatter()
+        ip.display_formatter.formatters["text/plain"] = formatter
+        formatter.for_type(object, func=display_hook)
+    except:
```

Comment:
This should probably be `except Exception` so as not to catch any system exceptions.